### PR TITLE
[DBEX] add enum, methods and scopes to Form526Submission

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

- Adds to `Form526Submission` model (utility described in [526 State Repair Technical Design Document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/untouched_submission_audit/526_state_repair_tdd.md#b-add-backup_submitted_claim_status-enum-to-the-form526submission-model))
  - `backup_submitted_claim_status` enum with 3 allowed values: `:accepted`, `:rejected` and `nil` (default)
  - instance methods: `remediated?`, `success_type?`, `in_process?`, `failure_type?`, `duplicate?`
  - scopes: `success_type`, `in_process`, `failure_type`
- *This work is behind a feature toggle (flipper): ~YES~/**NO***
- Responsible team: Disability Benefits Experience Team 2 (dBeX Carbs 🥖)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-api/pull/17254
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86438
- [526 State Repair Technical Design Document
](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/untouched_submission_audit/526_state_repair_tdd.md#b-add-backup_submitted_claim_status-enum-to-the-form526submission-model)

## Testing done

- [x] *New code is covered by unit tests*
- Add to `form526_submission_spec`:
  - `backup_submitted_claim_status` validations
  - instance method tests
  - scope tests
- Create `form526_submission_remediation` factory
- Add traits to `form526_submission` factory

## What areas of the site does it impact?

Database records are never intended to be created by application code; they will be created manually by a developer, most likely via a script at the time of remediation.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/form526_submission.rb [hunk 1]
```diff
@@ -157,12 +157,29 @@ class Form526Submission < ApplicationRecord
...
+  enum backup_submitted_claim_status: { accepted: 0, rejected: 1 }
   enum submit_endpoint: { evss: 0, claims_api: 1, benefits_intake_api: 2 }
 
   scope :pending_backup_submissions, lambda {
     where(aasm_state: 'delivered_to_backup')
       .where.not(backup_submitted_claim_id: nil)
   }
+  scope :in_process, lambda {
+    where(submitted_claim_id: nil, backup_submitted_claim_status: nil)
+      .where('created_at >= ?', 1.day.ago)
```
Comment (SamStuckey) [inline app/models/form526_submission.rb:169]: this should probably be more than a day, to account for long running back up submissions. Also we should make it a constant

File: app/models/form526_submission.rb [hunk 2]
```diff
@@ -157,12 +157,29 @@ class Form526Submission < ApplicationRecord
   belongs_to :user_account, dependent: nil, optional: true
 
   validates(:auth_headers_json, presence: true)
+  enum backup_submitted_claim_status: { accepted: 0, rejected: 1 }
```
Comment (SamStuckey) [inline app/models/form526_submission.rb:160]: will this allow `nil`?  I needs to, if not

File: app/models/form526_submission.rb [hunk 3]
```diff
@@ -518,6 +535,31 @@ def lighthouse_validation_errors
...
+  end
+
+  def success_type?
+    (submitted_claim_id && backup_submitted_claim_status.nil?) ||
+      (backup_submitted_claim_id && accepted?) || remediated?
+  end
+
+  def in_process?
+    if submitted_claim_id.nil? && backup_submitted_claim_status.nil?
+      created_at >= 1.day.ago
```
Comment (SamStuckey) [inline app/models/form526_submission.rb:557]: again, use the scope and check for inclusion

File: app/models/form526_submission.rb [hunk 4]
```diff
@@ -620,4 +662,8 @@ def submit_flashes
   def cleanup
     EVSS::DisabilityCompensationForm::SubmitForm526Cleanup.perform_async(id)
   end
+
+  def last_remediation
+    form526_submission_remediations&.order(created_at: :desc)&.last
```
Comment (SamStuckey) [inline app/models/form526_submission.rb:667]: I think we want `:asc` if using `.last`? 🤔

**Review by SamStuckey — CHANGES_REQUESTED** [review]
a couple notes, amazing specs btw!

File: app/models/form526_submission.rb (hunk 2 above)
Comment (freeheeling) [reply app/models/form526_submission.rb:160]: Yes, an instance's `backup_submitted_claim_status` attribute value is not prevented from being `nil`.

File: app/models/form526_submission.rb (hunk 4 above)
Comment (freeheeling) [reply app/models/form526_submission.rb]: Good catch! Corrected in [ccd6694](https://github.com/department-of-veterans-affairs/vets-api/pull/17274/commits/ccd66940f6df82e0f004b5fdce9c60a6bda9be70).

File: app/models/form526_submission.rb [hunk 5]
```diff
@@ -518,6 +535,31 @@ def lighthouse_validation_errors
...
+
+  def remediated?
+    last_remediation&.success || false
+  end
+
+  def failure_type?
+    !success_type? && !in_process?
+  end
+
+  def success_type?
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply app/models/form526_submission.rb:551]: Updated in [ccd6694](https://github.com/department-of-veterans-affairs/vets-api/pull/17274/commits/ccd66940f6df82e0f004b5fdce9c60a6bda9be70).

File: app/models/form526_submission.rb (hunk 3 above)
Comment (freeheeling) [reply app/models/form526_submission.rb]: Updated in [ccd6694](https://github.com/department-of-veterans-affairs/vets-api/pull/17274/commits/ccd66940f6df82e0f004b5fdce9c60a6bda9be70).

File: app/models/form526_submission.rb (hunk 1 above)
Comment (freeheeling) [reply app/models/form526_submission.rb]: Increased to 7 days and made constant in [ccd6694](https://github.com/department-of-veterans-affairs/vets-api/pull/17274/commits/ccd66940f6df82e0f004b5fdce9c60a6bda9be70).

**Comment by freeheeling** [comment]
> a couple notes, amazing specs btw!

Addressed. Appreciate the review and suggestions.

**Review by SamStuckey — APPROVED** [review]
👏 👏 👏

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**